src/lib/context.ts

```diff
@@ -95,24 +105,26 @@ export class AppContext {
         }
     }
 
-    log(message: string, extra: Dict = {}) {
-        this.logger.info(this.prepareExtra(extra), this.prepareLogMessage(message));
-        this.span?.log(Object.assign({}, this.utils.redactSensitiveKeys(extra), {event: message}));
+    log(message: string, extra?: Dict) {
+        const preparedExtra = this.prepareExtra(this.mergeExtra(this.loggerExtra, extra));
+
+        this.logger.info(preparedExtra, this.prepareLogMessage(message));
+        this.span?.log(Object.assign({}, preparedExtra, {event: message}));
     }
 
-    logError(message: string, error?: AppError | Error | unknown, extra: Dict = {}) {
+    logError(message: string, error?: AppError | Error | unknown, extra?: Dict) {
         if (error) {
             this.logger.error(
-                Object.assign({}, extractErrorInfo(error), {
-                    extra: this.utils.redactSensitiveKeys(extra),
-                }),
+                Object.assign(
```

Comment:
Was this change of order intentional? 

Also, can you please add a few test cases that will check logic added in this PR?

Reply:
I do not insist, but I think it would be useful to save the error keys
Otherwise, error fields may be accidentally overridden by extra data with same keys